octavia/f5_extensions/workarounds.py

```diff
@@ -0,0 +1,40 @@
+#    Copyright 2020 SAP SE
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from  octavia.f5_extensions import exceptions
+from octavia.db import repositories
+from octavia.db import api as db_api
+
+def check_member_for_invalid_ip(member):
+    """When creating a pool member that has the same IP as any VIP in the same network it will lead to error messages
+    on the F5 BigIP device.
+    This function checks the IP address of the given member and raises an exception if it finds a conflict.
+
+    Note that it doesn't suffice to use this workaround only when creating a member. It also needs to be used when
+    updating/bulk-updating members and VIP creation.
+    """
+
+    # get subnet containing member
+    member_ip = member.address
+    member_subnet = member.subnet_id
+
+    # get VIPs from network
+    try:
+        conflicts = repositories.VipRepository().get(db_api.get_session(autocommit=False), subnet_id=member_subnet)
```

Comment:
This just checks if a VIP from the same subnet is used. This will be probably always true and block like every member call.

Reply:
Also, please keep in mind that _any_ vip ip_address in the same network should be prohibited, not only from the same subnet. the subnet_id is an optional member property.

Reply:
Those two comments sound paradoxical. Same VIP from the same subnet should be allowed but not from the same network?
I suppose you only mean the latter.